src/wallet/coinselection.cpp

```diff
@@ -395,3 +382,54 @@ CAmount GetSelectionWaste(const std::set<CInputCoin>& inputs, CAmount change_cos
 
     return waste;
 }
+
+void SelectionResult::ComputeAndSetWaste(CAmount change_cost)
+{
+    m_waste = GetSelectionWaste(m_selected_inputs, change_cost, m_target, m_use_effective);
+}
+
+CAmount SelectionResult::GetWaste() const
+{
+    Assume(m_waste != std::nullopt);
+    return *m_waste;
+}
+
+CAmount SelectionResult::GetSelectedValue() const
```

Comment:
In 1b1ad425d6b2ca951e0a7e6096819fc5e468af4b: [[nodiscard]] the `GetWaste()` and `GetSelectedValue()` functions?

Reply:
Done